client/coral-admin/src/routes/Moderation/components/Moderation.js

```diff
@@ -30,6 +42,19 @@ export default class Moderation extends Component {
     key('d', this.moderate(true));
   }
 
+  componentWillUpdate() {
+    const {router, route, root: {asset, reportedCount}} = this.props;  
+
+    // Will update the URL to premod or new based on the moderation of the *asset*
+    // If there is a reported comments, reported queue has priority!
+    if (route.path === ':id') {
+      const queue = isPremod(asset.settings.moderation) ? 'premod' : 'new';
+      router.push(
+        getModPath(reportedCount ? 'reported' :  queue, asset.id)
+      );
+    }
+  }
```

Comment:
This one is necessary when there is a transition that updates. i.e clicking the StorySearch results